Src/MakingSense.SafeBrowsing/GoogleSafeBrowsing/GoogleSafeBrowsingUpdater.cs

```diff
@@ -0,0 +1,141 @@
+        private readonly GoogleSafeBrowsingDatabase _database;
+        private readonly SafebrowsingService _safebrowsingService;
+
+        private const string ANY_PLATFORM = "ANY_PLATFORM";
+        private const string URL = "URL";
+        private const string RAW = "RAW";
+        private const string FULL_UPDATE = "FULL_UPDATE";
+
+        /// <summary>
+        /// Creates new instance and initializes configuration parameters
+        /// </summary>
+        /// <param name="apiKey">Google API KEY</param>
+        /// <param name="clientId">Unique client ID</param>
+        /// <param name="clientVersion">Client version</param>
+        /// <param name="region">Geographic location in ISO 3166-1 alpha-2 format</param>
+        public GoogleSafeBrowsingUpdater(string apiKey, string clientId, string clientVersion, string region = null)
+        {
+            _configuration = new GoogleSafeBrowsingConfiguration
+            {
+                ApiKey = apiKey,
+                ClientId = clientId,
+                ClientVersion = clientVersion,
+                Region = region
+            };
+
+            _safebrowsingService = new SafebrowsingService(new BaseClientService.Initializer
+            {
+                ApiKey = _configuration.ApiKey,
+            });
+
+            _database = new GoogleSafeBrowsingDatabase();
+        }
+
+        /// <summary>
+        /// Updates Google Safe Browsing lists based on the remote resource
+        /// </summary>
+        /// <returns></returns>
+        public async Task UpdateAsync()
+        {
+            if (!_database.AllowRequest)
+            {
+                return;
+            }
+
+            var request = new FetchThreatListUpdatesRequest
+            {
+                Client = new ClientInfo
+                {
+                    ClientId = _configuration.ClientId,
+                    ClientVersion = _configuration.ClientVersion
+                },
+                ListUpdateRequests = _database.SuspiciousLists.Select(list => new ListUpdateRequest
+                {
+                    PlatformType = ANY_PLATFORM,
+                    ThreatType = list.Key,
+                    ThreatEntryType = URL,
+                    State = list.Value.State,
+                    Constraints = new Constraints
+                    {
+                        Region = _configuration.Region,
+                        SupportedCompressions = new List<string>
+                    {
+                        RAW
+                    }
+                    }
+                }).ToList()
+            };
+
+            var response = await _safebrowsingService.ThreatListUpdates.Fetch(request).ExecuteAsync();
+
+            _database.MinimumWaitDuration = TimeSpan.FromSeconds(double.Parse((response.MinimumWaitDuration as string).TrimEnd('s')));
+            _database.Updated = DateTimeOffset.Now;
+
+            foreach (var listUpdate in response.ListUpdateResponses)
```

Comment:
I think that this block could be easier to read if it is expressed as Linq expression.